### DISCUSSION TITLE
Oqtane and SEO

### DISCUSSION THREAD

**Discussion by mixlplix** [oqtane/oqtane.framework#2428] [Q&A]
We are putting together a front end product search page for a client.  Ironically, their existing product search uses DNN to dynamically build meta data for SEO purposes.  I'm wondering if there is an existing feature for Oqtane?

**Comment by sbwalker** [comment]
You will need to explain more about your use case. Oqtane does have properties you can set for Pages which are injected during the initial server-side page render as well as dynamically on client-side navigation using JS Interop.

**Comment by mixlplix** [comment]
Hi Shaun, the documentation is a bit sparse, is there an article you can point me to?  Specifically for this: “Oqtane does have properties you can set for Pages which are injected during the initial server-side page render”

**Comment by sbwalker** [comment]
@mixlplix SEO is only applicable to bots and crawlers - so basically only the initial server-side page render scenario is relevant (not subsequent client-side navigations). You need to ensure you have Prerendering enabled for your site in Site Settings so that the module titles and content are indexable. In regards to page head information, the Page Management UI which contains Title and Meta fields in the Appearance section (poorly named admittedly):

![image](https://user-images.githubusercontent.com/4840590/192663984-bb295741-11f9-4ab4-afbf-c5820c2c1237.png)

The information entered here is injected into the page during the initial server-side page render (in Oqtane.Server/Pages/_host.cshtml). It is not currently extensible (ie. developers cannot dynamically affect these values at runtime). Server-side extensibility of page output has never been a focus of Oqtane - the focus has always been on client-side extensibility via modules, themes, etc...

**Comment by thienvc** [comment]
Oqtane now not optimize for Google SEO
![image](https://user-images.githubusercontent.com/1115393/198920473-18f773c7-24ce-427b-8ebe-37130980b7ed.png)

**Comment by iJungleboy** [comment]
@thienvc just FYI: I believe Oqtane by itself isn't there yet, but I think we have it solved in 2sxc (an Oqtane plugin). 

In the next few days we'll launch a website showing how to create awesome websites with Oqtane and then do a bunch of tests to verify everything. It's not ready yet, but you can already have a preview on https://blazor-cms.org/ 💖
